src/drivers/intel/dmic/dmic_nhlt.c

```diff
@@ -272,22 +278,24 @@ int dmic_set_config_nhlt(struct dai *dai, void *spec_config)
 	/* Write the FIFO control registers. The clear/set of bits is the same for
 	 * all DMIC_HW_VERSION
 	 */
+	/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
+	val = (out_control[dai->index] &
+		~(OUTCONTROL0_TIE_BIT | OUTCONTROL0_SIP_BIT | OUTCONTROL0_FCI_BIT)) |
+		OUTCONTROL0_FINIT_BIT;
 	if (dai->index == 0) {
-		/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
-		val = (out_control[0] &
-			~(OUTCONTROL0_TIE_BIT | OUTCONTROL0_SIP_BIT | OUTCONTROL0_FCI_BIT)) |
-			OUTCONTROL0_FINIT_BIT;
 		dai_write(dai, OUTCONTROL0, val);
-		dai_dbg(dai, "dmic_set_config_nhlt(): OUTCONTROL0 = %08x", val);
+		dai_info(dai, "dmic_set_config_nhlt(): OUTCONTROL0 = %08x", val);
 	} else {
-		/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
-		val = (out_control[1] &
-			~(OUTCONTROL1_TIE_BIT | OUTCONTROL1_SIP_BIT | OUTCONTROL1_FCI_BIT)) |
-			OUTCONTROL1_FINIT_BIT;
 		dai_write(dai, OUTCONTROL1, val);
-		dai_dbg(dai, "dmic_set_config_nhlt(): OUTCONTROL1 = %08x", val);
+		dai_info(dai, "dmic_set_config_nhlt(): OUTCONTROL1 = %08x", val);
 	}
```

Comment:
Is there a reason to change to dai_info?
At the same time this can be simpler, if I'm not mistaken:
```c
/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
val = (out_control[dai->index] &
	~(OUTCONTROL0_TIE_BIT | OUTCONTROL0_SIP_BIT | OUTCONTROL0_FCI_BIT)) |
	OUTCONTROL0_FINIT_BIT;

if (dai->index == 0)
	dai_write(dai, OUTCONTROL0, val);
else
	dai_write(dai, OUTCONTROL1, val);

dai_dbg(dai, "%s(): OUTCONTROL%d = %08x", __func__, dai->index, val);
```

Reply:
I might be wrong, but out_control[1] is never actually got assigned a value in the loop at line (232 / 238) because at line 227 / 233 you just only allow num_fifos to be exactly 1?

AM I missing something?

Reply:
> s there a reason to change to dai_info?

This is one of the most common things how the blob could be set up wrong and dai_dbg() does not work in my opinion. I've never been able to see DAI debug traces with sof-logger. SSP also outputs some key settings as info. If that's possible, I'll of course change this.

Reply:
> I might be wrong, but out_control[1] is never actually got assigned a value in the loop at line (232 / 238) because at line 227 / 233 you just only allow num_fifos to be exactly 1?
> 
> Am I missing something?

Please ignore this, num_fifos can be 1 or 2.